src/selectors/entities/sidebar.js

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {General, Preferences} from 'constants/index';
+import {getConfig} from 'selectors/entities/general';
+import {getBool as getBoolPreference, get as getPreference} from 'selectors/entities/preferences';
+
+const defaultSidebarPrefs = {
+    grouping: 'by_type',
+    unreads_at_top: 'true',
+    favorite_at_top: 'true',
+    sorting: 'alpha',
+};
+
+export function getSidebarPreferences(state) {
```

Comment:
Any reason this isn't a selector?

Reply:
It looks like it should probably be a selector since it's returning an object. Something like this should work
```javascript
export const getSidebarPreferences = createSelector(
    (state) => {
        const config = getConfig(state);
        return config.ExperimentalGroupUnreadChannels !== General.DISABLED && getBoolPreference(
            state,
            Preferences.CATEGORY_SIDEBAR_SETTINGS,
            'show_unread_section',
            config.ExperimentalGroupUnreadChannels === General.DEFAULT_ON
       );
    },
    (state) => getPreference(state, Preferences.CATEGORY_SIDEBAR_SETTINGS, '', null),
    (showUnreadSection, sidebarPreferencesJson) => {
        // ...
    }
);
```

Reply:
Thanks for the feedback.  I'll look at making this a selector